### DISCUSSION TITLE
Current calibration for inaccurate BMS

### DISCUSSION THREAD

**Discussion by Karatzie** [fl4p/batmon-ha#121] [Ideas]
Hi fl4p.

Thank you for your great Home Assistant addon.

I have connected two Supervolt batteries in parallel.
Each battery can provide 320 A for 5 seconds.
When I need a very little power in the motorhome, the batteries show no consumption.
Only standby is displayed and the BMS does not notice the discharge.
Only from 400 mA consumption per battery is the discharge recognised and the remaining capacity is calculated correctly. 
However, the BMS only measures 300 mA. 
The calculation of the remaining capacity and the expected discharge time are not correct.

In the settings, we can adjust the sign with current_calibration.
Here you could also insert a filter with calibrate linear.
Example:
filters:
  - offset: 2.0
  - multiply: 1.2
  - calibrate_linear:
      - 0.0 -> 0.4
      - 0.3 -> 0.42
      - 1.0 -> 1.1
      - 100.0 -> 100.0

I'm waiting for your Supervolt integration and how things are going with your Batmon.

Best regards
Andreas

**Comment by fl4p** [comment]
Hi Andreas, I merged the Supervolt code.

Put the multi point calibration on the list
